script/release.js

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env node
+
+const { versionBump, currentVersion }  = require('./version-utils');
+const { logger, spawnOrFail, prompt, shouldContinuePrompt, quit, process, path } = require('./cli-utils');
+
+const deployDemo = (version) => {
+  const demo_name = `chime-sdk-demo-${version.replace(/\./g, "-")}`;
+  logger.log(`Deploying ${demo_name} ...`);
+  process.chdir(path.join(__dirname, '../demos/serverless'));
+  spawnOrFail('npm', [`run deploy -- -b ${demo_name} -s ${demo_name} -o ${demo_name} -u false`], { printErr: true });
+};
+
+const getCurrentRemoteBranch = () => {
+  return (spawnOrFail('git', ['for-each-ref --format="%(upstream:short)" "$(git symbolic-ref -q HEAD)"'], { skipOutput: true })).trim();
+};
+
+const buildAndPack = () => {
+  logger.log('Building package...');
+  spawnOrFail('npm', ['run build:release']);
+  logger.log('Packaging ...');
+  spawnOrFail('npm', ['pack --dry-run'], { printErr: true });
+};
+
+const cleanUp = async (remoteBranch) => {
+  logger.warn(`Warning: Resetting HEAD ${remoteBranch ? `to ${remoteBranch}` : ''}.\nAll current staged and local changes will be lost.`);
+  await shouldContinuePrompt();
+  spawnOrFail('git', [`reset --hard ${remoteBranch ? remoteBranch : ''}`]);
+  spawnOrFail('git', [' clean -ffxd .']);
+};
+
+const release = async () => {
+  spawnOrFail('git', ['fetch origin'], { skipOutput: true });
+  const currentBranch = (spawnOrFail('git', [' branch --show-current'], { skipOutput: true })).trim();
+  const remoteBranch = getCurrentRemoteBranch();
+  if (!remoteBranch || (remoteBranch !== 'origin/main' && (/^origin\/release-[0-9]+\.x$/).test(remoteBranch))) {
+    logger.error(`The local branch ${currentBranch} does not track either main or release-<version>.x branch`);
+    quit(1);
+  }
+  await cleanUp(remoteBranch);
+
+  buildAndPack();
+  logger.log('Do you want to upload these files to release branch?\n');
+  await shouldContinuePrompt();
+  spawnOrFail('git', ['push origin HEAD:release -f']);
+  deployDemo(currentVersion);
+
+  //Bump next development version
+  await versionBump();
```

Comment:
There are no params here. So, I guess you want the user to follow the prompts and pass the param for what type of release they want to do. 

It will be nice if we can pass those params to the release script itself. Is that something you plan to add?

Reply:
i pass param for hotfix workflow to auto increase to patch version. For release, it is a scripted diaglog so not sure the use case here you refer here. If needed, we can always pass param like in hotfix.